src/test/java/com/google/cloud/pontem/AvroUtilTest.java

```diff
@@ -40,11 +40,30 @@ public void testGetSingleAvroTypeFromNullableUnion() {
             SchemaBuilder.builder().nullable().stringType()));
   }
 
+  @Test
+  public void testGetSingleAvroTypeFromNullableUnion2() {
+    assertEquals(
+        SchemaBuilder.builder().stringType(),
+        AvroUtil.getSingleAvroTypeFromNullableUnion(
+            SchemaBuilder.builder()
+                .unionOf()
+                .stringType()
+                .and()
+                .type(Schema.create(Schema.Type.NULL))
+                .endUnion()));
+  }
+
   @Test(expected = IllegalArgumentException.class)
   public void testGetSingleAvroTypeFromNullableUnion_exception() {
     AvroUtil.getSingleAvroTypeFromNullableUnion(SchemaBuilder.builder().stringType());
   }
 
+  @Test(expected = IllegalArgumentException.class)
+  public void testGetSingleAvroTypeFromNullableUnion_exception_null() {
```

Comment:
It's not entirely clear to me how and where the _null suffix comes in to play here can this be made more explicit?